lib/hal_api/controller/exceptions.rb

```diff
@@ -8,10 +8,18 @@ def respond_with_error(exception)
     wrapper = ::ActionDispatch::ExceptionWrapper.new(env, exception)
     log_error(env, wrapper)
 
-    error = exception.is_a?(HalApi::Errors::ApiError) ? exception : HalApi::Errors::ApiError.new
+    error = if exception.is_a?(HalApi::Errors::ApiError)
+              exception
+            else
+              e = HalApi::Errors::ApiError.new(exception.message)
+              e.set_backtrace(exception.backtrace)
+              e
+            end
+
     respond_with(
       error,
       status: error.status,
+      location: nil, # for POST requests
```

Comment:
oh, now that looks familiar, I have seen this before - and thought I fixed that.
perhaps the create_options is insufficient or should be setting to nil instead of ''?